File: redsheet-portal/src/app/database-models/negotiation-type.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectBackground,  } from './'
export class NegotiationType {
    constructor(negotiationType?: NegotiationType )  {
        let properties = [ "className", "negotiationTypeId", "negotiationTypeName", "projectBackgrounds",];
        for (let property of properties)
            if (negotiationType && negotiationType[property])
                this[property] = negotiationType[property];
    }
 
    @required()
    @maxLength(200)
	className : string =   undefined;
 
	negotiationTypeId : number =   0 ;
 
    @required()
    @maxLength(200)
	negotiationTypeName : string =   undefined;
	@nested(ProjectBackground)
	projectBackgrounds: ProjectBackground[];



}

File: RedSheet.DbEntities/Models/NegotiationType.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("NegotiationTypes", Schema = "dbo")]
	[Serializable]
	public partial class NegotiationType  
	{
 
	[Required] 
	[MaxLength(200)]
	public string ClassName { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int NegotiationTypeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string NegotiationTypeName { get; set; } 
	[InverseProperty("NegotiationType")]
	public ICollection<ProjectBackground> ProjectBackgrounds { get; set; } 


	public NegotiationType()
	{
	this.ProjectBackgrounds = new HashSet<ProjectBackground>();

    }

    }
}

